urllib3/request.py

```diff
@@ -149,3 +187,24 @@ def request_encode_body(self, method, url, fields=None, headers=None,
         extra_kw.update(urlopen_kw)
 
         return self.urlopen(method, url, **extra_kw)
+
+    def redirect(self, response, method, retries, **kwargs):
+        """
+        Abstracts the redirect process to be used from any :class:`RequestMethods` object
+        """
+        url = kwargs.pop('url', '')
+        redirect_location = urljoin(url, response.get_redirect_location())
+        method = retries.redirect_method(method, response.status)
+        try:
+            pool = kwargs.pop('pool', self)
```

Comment:
I'm not disputing that `MaxRetryError` can be raised, I'm saying that `dict.pop` can't raise it. =)